src/anon-socks-client.ts

```diff
@@ -0,0 +1,56 @@
+import axios, { AxiosInstance, AxiosRequestConfig, AxiosResponse } from 'axios';
+import { SocksProxyAgent } from 'socks-proxy-agent';
+import { Anon } from './anon';
+
+export class AnonSocksClient {
+  private anon?: Anon;
+  private agent: SocksProxyAgent;
+  public axios: AxiosInstance;
+  private socksPort: number;
+
+  constructor(anon: Anon);
+  constructor(socksPort: number);
+  constructor(anonOrPort: Anon | number) {
+    if (anonOrPort instanceof Anon) {
+      this.anon = anonOrPort;
+      this.socksPort = this.anon.getSOCKSPort();
+    } else {
+      this.socksPort = anonOrPort;
+    }
+
+    if (this.socksPort === 0) {
+      throw new Error('Invalid SOCKS port: 0. SOCKS proxy must be enabled with a valid port number (1-65535).');
+    }
+    
+    this.agent = this.createAgent();
+    this.axios = axios.create({
+      httpAgent: this.agent,
+      httpsAgent: this.agent
+    });
+  }
+
+  private createAgent(): SocksProxyAgent {
+    return new SocksProxyAgent(`socks://127.0.0.1:${this.socksPort}`);
```

Comment:
move host to property to be able to configure it in the future or now, if not localhost